### ISSUE TITLE
ML: Address class imbalance in HybridGeneFilter (CNN+Dense) — add pos_weight loss and retrain

### ISSUE THREAD

**Issue by roeimed0** [roeimed0/bacterial-gene-prediction#112]
## Problem

`HybridGeneFilter` uses a CNN + Dense hybrid model trained with `BCEWithLogitsLoss`. The ORF-level training data has a ~44:1 negative:positive ratio, but `pos_weight` is never passed to the loss. The threshold was manually lowered to 0.12 post-hoc, indicating the imbalance was noticed but never addressed at the training level.

## What exists already

- `HybridGenePredictor` — full CNN+Dense architecture defined in `ml_models.py`
- `extract_features()` + `_one_hot_encode_dna()` — feature and sequence encoding ready to use for training
- Trained model stored as `models/hybrid_best_model.pkl` (pickle, contains `model_state_dict` + `threshold`)

### 1. Add a `train()` method to `HybridGeneFilter`

```python
def train(self, candidates: list, labels: np.ndarray, ...):
    n_pos = labels.sum()
    n_neg = len(labels) - n_pos
    pos_weight = torch.tensor([n_neg / n_pos], device=self.device)
    criterion = nn.BCEWithLogitsLoss(pos_weight=pos_weight)
    # standard training loop with early stopping
```

Consider **focal loss** as an alternative to `pos_weight` — focal loss down-weights easy negatives via `-(1-p)^γ * log(p)`, which is better suited for 44:1 ratios where the majority of negatives are confidently-classified easy cases.

### 2. Add PR-curve threshold selection

After training, find the F1-maximising threshold on a validation set instead of hardcoding 0.12.

### 3. Extend `scripts/train_models.py` (from the LightGBM issue)

Add a HybridGeneFilter training step after the LightGBM step, using the same labeled ORF dataset.

## Expected impact

The CNN sees sequence context that LightGBM does not — fixing its imbalance bias could meaningfully improve recall on Archaea and organisms with non-canonical gene features where the sequence signal matters most.

## Related

- Companion issue: OrfGroupClassifier (LightGBM) retraining
- Supersedes #98

**Comment by roeimed0** [comment]
Implemented and validated. HybridGeneFilter retrained on output of new LGB model with BCEWithLogitsLoss(pos_weight). +4.24pp mean F1 across 20 held-out genomes, +3.65pp sensitivity, +2.74pp precision. 19/20 genomes improved. Threshold set to 0.25 based on end-to-end F1 sweep.
